tensorcircuit/results/readout_mitigation.py

```diff
@@ -337,10 +343,73 @@ def apply_readout_mitigation(self, raw_count: ct, method: str = "inverse") -> ct
         probability = probability * shots
         return vec2count(probability, prune=True)
 
+    def mapping_preprocess(
+        self,
+        counts: ct,
+        qubits: Sequence[int],
+        positional_logical_mapping: Optional[Dict[int, int]] = None,
+        logical_physical_mapping: Optional[Dict[int, int]] = None,
+    ) -> ct:
+        """
+        Preprocessing to deal with qubit mapping, including positional_logical_mapping and
+        logical_physical_mapping. Return self.use_qubits(physical) and corresponding counts.
+
+        :param counts: raw_counts on positional_qubits
+        :type counts: ct
+        :param qubits: user-defined logical qubits to show final mitted results
+        :type qubits: Sequence[int]
+        :param positional_logical_mapping: positional_logical_mapping, defaults to None.
+        :type positional_logical_mapping: Optional[Dict[int, int]], optional
+        :param logical_physical_mapping: logical_physical_mapping, defaults to None
+        :type logical_physical_mapping: Optional[Dict[int, int]], optional
+        :return: counts on self.use_qubit(physical)
+        :rtype: ct
+        """
+        # counts [0,1,2] / logic_qubit(circuit,meas) [3,4,6] / physic_qubit [2,8,6]
+        # / use_logic_qubits [4,3] / cal_qubits[0-8]
+        # input: counts [0,1,2], use_logical_qubits [4,3],  logic_physic_mapping {3:2,4:8,6:6},
+        # input: position_logical_mapping{0:3,1:4,2:6}
+        # self.use_qubits(physic)[8,2]
+        # use_position_qubits [1,0]
+        # counts = marginal_count(counts[0,1,2], [1,0])  corresponds to self.use_qubits(physic)
+
+        if not is_sequence(qubits):
+            qubits = list(range(qubits))  # type: ignore
+
+        if logical_physical_mapping is None:
+            self.use_qubits = qubits  # type: ignore
+
+            if positional_logical_mapping is None:
+                use_position_qubits = self.use_qubits
+            else:
+                logical_qubits = list(positional_logical_mapping.values())
+                use_position_qubits = []
+                for i in qubits:
+                    use_position_qubits.append(logical_qubits.index(i))
```

Comment:
reduce L385-L388 to 
``use_position_qubits = [logical_positional_mapping[lq] for lq in qubits]``